components/eam/src/chemistry/utils/read_spa_data.F90

```diff
@@ -0,0 +1,93 @@
+module read_spa_data
+
+!BSINGH - I didn't add any comments here due to time constrain but I will add comments later
+
+use tracer_data
+
+implicit none
+
+private
+
+public :: read_spa_data_init, read_spa_data_adv, read_spa_data_register
+
+type(trfld), pointer :: spa_fields(:)
+type(trfile)         :: spa_file
+logical              :: rmv_file = .false.
+
+!fields to read from the file
+character(len=16), parameter :: pbuf_names(59) = &
+     [ 'AER_G_SW_0   ', 'AER_G_SW_1   ', 'AER_G_SW_2   ', 'AER_G_SW_3   ', 'AER_G_SW_4   ', &
+       'AER_G_SW_5   ', 'AER_G_SW_6   ', 'AER_G_SW_7   ', 'AER_G_SW_8   ', 'AER_G_SW_9   ', &
+       'AER_G_SW_10  ', 'AER_G_SW_11  ', 'AER_G_SW_12  ', 'AER_G_SW_13  ', &
+       'AER_SSA_SW_0 ', 'AER_SSA_SW_1 ', 'AER_SSA_SW_2 ', 'AER_SSA_SW_3 ', 'AER_SSA_SW_4 ', &
+       'AER_SSA_SW_5 ', 'AER_SSA_SW_6 ', 'AER_SSA_SW_7 ', 'AER_SSA_SW_8 ', 'AER_SSA_SW_9 ', &
+       'AER_SSA_SW_10', 'AER_SSA_SW_11', 'AER_SSA_SW_12', 'AER_SSA_SW_13', &
+       'AER_TAU_LW_0 ', 'AER_TAU_LW_1 ', 'AER_TAU_LW_2 ', 'AER_TAU_LW_3 ', 'AER_TAU_LW_4 ', &
+       'AER_TAU_LW_5 ', 'AER_TAU_LW_6 ', 'AER_TAU_LW_7 ', 'AER_TAU_LW_8 ', 'AER_TAU_LW_9 ', &
+       'AER_TAU_LW_10', 'AER_TAU_LW_11', 'AER_TAU_LW_12', 'AER_TAU_LW_13', 'AER_TAU_LW_14', &
+       'AER_TAU_LW_15', &
+       'AER_TAU_SW_0 ', 'AER_TAU_SW_1 ', 'AER_TAU_SW_2 ', 'AER_TAU_SW_3 ', 'AER_TAU_SW_4 ', &
+       'AER_TAU_SW_5 ', 'AER_TAU_SW_6 ', 'AER_TAU_SW_7 ', 'AER_TAU_SW_8 ', 'AER_TAU_SW_9 ', &
+       'AER_TAU_SW_10', 'AER_TAU_SW_11', 'AER_TAU_SW_12','AER_TAU_SW_13', &
+       'CCN3         ']
+
+character(len=16), parameter :: specifier(59) = pbuf_names(59)
```

Comment:
May be get rid of `pbuf_names` variable and use `specifier` only everywhere or vice versa.

Reply:
Maybe this is just what you're saying, but it looks to me like you created a char vector called pbuf_names then created an identical copy of it called specifier. Both are parameter types so can't be changed, so why do that?